lib/Sema/TypeCheckDeclPrimary.cpp

```diff
@@ -621,6 +621,60 @@ CheckRedeclarationRequest::evaluate(Evaluator &eval, ValueDecl *current) const {
           continue;
       }
 
+      // Signatures are the same, but interface types are not. We must
+      // have a type that we've massaged as part of signature
+      // interface type generation. If it's a result of remapping a
+      // function parameter from 'inout T!' to 'inout T?'
```

Comment:
`inout`-ness does not matter here. This diagnostic applies to plain parameters as well.